dandi/metadata.py

```diff
@@ -270,9 +270,25 @@ def extract_wasDerivedFrom(metadata):
     models.Participant, "identifier", "subject_id", id=...
 )
 
-extract_wasGeneratedBy = extract_model_list(
-    models.Session, "name", "session_id", id=...
-)
+
+def extract_session(metadata):
+    probe_ids = metadata.get("probe_ids", [])
+    if isinstance(probe_ids, str):
+        probe_ids = [probe_ids]
+    probes = []
+    for val in probe_ids:
+        probes.append(models.Agent(identifier=f"probe:{val}", name="Ecephys Probe"))
+    if len(probes) == 0:
+        probes = None
```

Comment:
```suggestion
    probes = probes or None
```
or 
```suggestion
    if not probes:
        probes = None
```
imho would be more pythonic